src/components/utils/onfidoApi.js

```diff
@@ -19,22 +19,15 @@ export const uploadLivePhoto = (data, token, onSuccess, onError) => {
   sendFile(endpoint, data, token, onSuccess, onError)
 }
 
-export const getLivenessChallenges = (onSuccess/*, onError*/) => {
-  /*
-  @todo, configure real request
-  const endpoint = `${process.env.ONFIDO_API_URL}/v2/liveness`
-  sendFile(endpoint, {}, token, onSuccess, onError)
-  */
-  return onSuccess([
-    {
-      type: 'moveHead',
-      value: 'left',
-    },
-    {
-      type: 'repeatDigits',
-      value: [1, 3, 4],
-    }
-  ])
+export const uploadLiveVideo = (data, token, onSuccess, onError) => {
+  const {
+    challenges: challenge,
+    id: challenge_id,
+    switchSeconds: challenge_switch_at
+  } = data.challengeData
+  const payload = { file: data.blob, challenge: JSON.stringify(challenge), challenge_id, challenge_switch_at }
```

Comment:
Do you need to `JSON.stringify` here? Surely this happens later on anyhow?

Reply:
No, it doesn't happen later on. We are posting the file as form data. If I stringify all the parameters in the payload (excluding file) I will get a `malformed request` error. The value of `challenge` needs to be JSON otherwise I will get an error saying that the value of challenge is not a valid JSON.

Reply:
Ah cool - makes sense, thanks